lib/firmata.js

```diff
@@ -1411,6 +1451,148 @@ Board.prototype.stepperStep = function(deviceNum, direction, steps, speed, accel
 };
 
 /**
+ * Asks the Arduino to configure a hardware or serial port.
+ * @param {number} port The serial port to use (HW_SERIAL1, HW_SERIAL2, HW_SERIAL3, SW_SERIAL0,
+ * SW_SERIAL1, SW_SERIAL2, SW_SERIAL3)
+ * @param {number} baud The baud rate of the serial port.
+ * @param {number} bytesToRead The number of bytes to read on each iteration of the main loop
+ * (not yet used)
+ * @param {number} rxPin [SoftwareSerial only] The RX pin of the SoftwareSerial instance
+ * @param {number} txPin [SoftwareSerial only] The TX pin of the SoftwareSerial instance
+ */
+
+Board.prototype.serialConfig = function(portId, baud, bytesToRead, rxPin, txPin) {
+  var softSerialPins = {};
+  var data = [
+    START_SYSEX,
+    SERIAL_MESSAGE,
+    SERIAL_CONFIG | portId,
+    baud & 0x007F,
+    (baud >> 7) & 0x007F,
+    (baud >> 14) & 0x007F,
+    bytesToRead & 0x007F,
+    (bytesToRead >> 7) & 0x007F
+  ];
+  if (rxPin) {
+    data.push(rxPin);
+  }
+  if (txPin) {
+    data.push(txPin);
+  }
+  data.push(END_SYSEX);
+  this.transport.write(new Buffer(data));
+};
+
+/**
+ * Write an array of bytes to the specified serial port.
+ * @param {number} portId The serial port to write to.
+ * @param {array} inBytes An array of bytes to write to the serial port.
+ */
+
+Board.prototype.serialWrite = function(portId, inBytes) {
+  var data = [
+    START_SYSEX,
+    SERIAL_MESSAGE,
+    SERIAL_WRITE | portId
+  ];
+  for (var i = 0, len = inBytes.length; i < len; i++) {
+    data.push(inBytes[i] & 0x007F);
+    data.push((inBytes[i] >> 7) & 0x007F);
+  }
+  data.push(END_SYSEX);
+  if (data.length > 0) {
+    this.transport.write(new Buffer(data));
+  }
+};
+
+/**
+ * Start continuous reading of the specified serial port. The port is checked for data each
+ * iteration of the main Arduino loop.
+ * {number} portId The serial port to start reading continuously.
+ */
+
+Board.prototype.serialStartReading = function(portId) {
```

Comment:
> `serialRead(portId, callback)` 

This is actually how I imagined it would be shaped.

I'm not sure about the rest, I have to think about it more and at the moment I just landed in seattle—so I'm basically useless right now ;)